src/drawers/drawer.ts

```diff
@@ -5,238 +5,130 @@
+  public useCanvas(canvas: d3.Selection<HTMLCanvasElement, any, any, any>) {
+    if (this._svgRootG != null) {
+      this._svgRootG.remove();
+      this._svgRootG = null;
     }
+    this._canvas = canvas;
   }
 
   /**
-   * Binds data to selection
-   *
-   * @param{any[]} data The data to be drawn
+   * Public for testing
    */
-  private _bindSelectionData(data: any[]) {
-    const dataElementsUpdate = this.selection().data(data);
-    const dataElements =
-      dataElementsUpdate
-        .enter()
-        .append(this._svgElementName)
-        .merge(dataElementsUpdate);
-    dataElementsUpdate.exit().remove();
-
-    this._applyDefaultAttributes(dataElements);
-  }
-
-  protected _applyDefaultAttributes(selection: SimpleSelection<any>) {
-    if (this._className != null) {
-      selection.classed(this._className, true);
-    }
+  public getSvgRootG(): d3.Selection<SVGElement, any, any, any> {
+    return this._svgRootG;
   }
 
   /**
-   * Draws data using one step
-   *
-   * @param{AppliedDrawStep} step The step, how data should be drawn.
+   * public for testing
    */
-  private _drawStep(step: Drawers.AppliedDrawStep) {
-    const selection = this.selection();
-    const colorAttributes = ["fill", "stroke"];
-    colorAttributes.forEach((colorAttribute) => {
-      if (step.attrToAppliedProjector[colorAttribute] != null) {
-        selection.attr(colorAttribute, step.attrToAppliedProjector[colorAttribute]);
-      }
-    });
-    step.animator.animate(selection, step.attrToAppliedProjector);
-    if (this._className != null) {
-      this.selection().classed(this._className, true);
-    }
-  }
-
-  protected _drawStepCanvas(data: any[], step: Drawers.AppliedDrawStep) {
-    Utils.Window.warn("canvas rendering not yet implemented on " + (<any> this.constructor).name);
-  }
-
-  private _appliedProjectors(attrToProjector: AttributeToProjector): AttributeToAppliedProjector {
-    const modifiedAttrToProjector: AttributeToAppliedProjector = {};
-    Object.keys(attrToProjector).forEach((attr: string) => {
-      modifiedAttrToProjector[attr] =
-        (datum: any, index: number) => attrToProjector[attr](datum, index, this._dataset);
-    });
-
-    return modifiedAttrToProjector;
+  public getCanvas(): d3.Selection<HTMLCanvasElement, any, any, any> {
+    return this._canvas;
   }
 
   /**
-   * Calculates the total time it takes to use the input drawSteps to draw the input data
-   *
-   * @param {any[]} data The data that would have been drawn
-   * @param {Drawers.DrawStep[]} drawSteps The DrawSteps to use
-   * @returns {number} The total time it takes to draw
+   * Removes this Drawer's renderArea
    */
-  public totalDrawTime(data: any[], drawSteps: Drawers.DrawStep[]) {
-    let delay = 0;
-    drawSteps.forEach((drawStep, i) => {
-      delay += drawStep.animator.totalTime(data.length);
-    });
-
-    return delay;
+  public remove() {
+    if (this._svgRootG != null) {
+      this._svgRootG.remove();
+    }
```

Comment:
Also another opportunity to delegate to a common interface between the drawer types

```typescript
interface IDrawer {
    remove: () => void;
    ...
}
```